src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2824,6 +2824,21 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             }
             break;
         }
+        case SMART_ACTION_SET_SCALE:
+        {
+            uint32 scaleInteger = e.action.setScale.scaleInteger;
+            uint32 scaleFraction = e.action.setScale.scaleFraction;
+            float scale = float(scaleInteger) + float(scaleFraction) / std::pow(10, std::floor(std::log10(float(scaleFraction ? scaleFraction : 1)) + 1));
```

Comment:
```suggestion
            float scale = float(scaleInteger) + (float(scaleFraction) / 100);
```

Wouldn't this do the same thing, but easier?

scaleInteger = 2
scaleFraction = 75

2 + (75 / 100) = 2 + 0.75

Reply:
idk I just copied from SET_MOVEMENT_SPEED which is action 136, iirc

Reply:
Ah, well I dont see a reason not to just divde by 100

Reply:
And I havent checked this, but do you have a check for scaleFraction being greater than 99?

Reply:
I will check today after I wake up

Reply:
Changed to a percentage-based system instead